File: frontend/src/services/participant.service.ts
/**
 * Participant service for handling participant-related API calls
 */

import { api } from '@/lib/api';

const BASE_URL = '/participants';

/**
 * Deletes a participant from a tournament
 * @param participantId - The participant ID
 * @returns void
 */
export const deleteParticipant = async (participantId: string): Promise<void> => {
  await api.delete(`${BASE_URL}/${participantId}`);
};

export const participantService = {
  deleteParticipant,
};

File: backend/src/main/java/com/tft/tournament/service/ParticipantService.java
package com.tft.tournament.service;

import java.util.UUID;

/**
 * Service pour la gestion des participants.
 *
 * @author Fethi Benseddik
 * @version 1.0
 * @since 2025
 */
public interface ParticipantService {

    /**
     * Supprime un participant d'un tournoi.
     * L'utilisateur doit être l'organisateur du tournoi, un admin du tournoi,
     * ou le participant lui-même.
     *
     * @param participantId identifiant du participant
     * @param userId        identifiant de l'utilisateur qui effectue la suppression
     */
    void deleteParticipant(UUID participantId, UUID userId);
}
